addons/hr_timesheet/models/timer_mixin.py

```diff
@@ -0,0 +1,83 @@
+from odoo import models, fields, api
+
+
+class TimerMixin(models.AbstractModel):
+    _name = 'timer.mixin'
+    _description = 'Timer Mixin'
+
+    timer_start = fields.Datetime(
+        "Timer Start",
+        default=None
+    )
+    timer_pause = fields.Datetime("Timer Last Pause")
+    timer = fields.Boolean(
+        string=" ",
+        compute="_compute_timer",
+        store=False,
+        help="This field is used for timer_toggle_button widget"
+    )
+
+    @api.depends('timer_start')
+    def _compute_timer(self) -> None:
+        for record in self:
+            if record.timer_start:
+                record.timer = True
+            else:
+                record.timer = False
+
+    def action_timer_start(self) -> None:
+        """ Action start the timer.
+
+            Start timer and search if another timer hasn't been launched.
+            If yes, then stop the timer before launch this timer.
+        """
+        self.ensure_one()
+        if not self.timer_start:
+            self.write({
+                'timer_start': fields.Datetime.now(),
+            })
+
+    def action_timer_stop(self):
+        """ Action stop the timer.
+
+            Before stop the timer, we compute the minutes spent with the timer
+
+            :return minutes_spent if the timer is started,
+                    otherwise return False
+        """
+        self.ensure_one()
```

Comment:
``` python
def action_timer_stop(self):
    """ Stop the timer and return the spent minutes since it started
        :return minutes_spent if the timer is started,
                otherwise return False
    """
    self.ensure_one()
    if not self.timer_start:
        return False
    self.write({'timer_start': False, 'timer_pause': False})
    return self._compute_minutes_spent()
```

Reply:
@tivisse Thanks, I rewrite this method, but I can't call the "_compute_minutes_spent" method after the "write" method because I need the value of 'timer_start' field before this "write". Therefore, I will let the local variable named 'minutes_spent'